salt/states/pip_state.py

```diff
@@ -163,8 +164,11 @@ def _check_pkg_version_format(pkg):
         ret['version_spec'] = []
     else:
         ret['result'] = True
-        ret['prefix'] = install_req.req.project_name
-        ret['version_spec'] = install_req.req.specs
+        if hasattr(install_req.req, 'project_name'):
+            ret['prefix'] = install_req.req.project_name
+        else:
+            ret['prefix'] = re.sub('[^A-Za-z0-9.]+', '-', install_req.req.name)
+        ret['version_spec'] = [(spec.operator, spec.version) for spec in install_req.req.specifier]
```

Comment:
And once this fix is in we will spend more time exploring how to best get away from the api completely.